src/S3/Input/AccessControlPolicy.php

```diff
@@ -56,4 +56,14 @@ public function setOwner(?Owner $value): self
 
         return $this;
     }
+
+    public function validate(): void
+    {
+        foreach ($this->Grants as $item) {
+            $item->validate();
+        }
+        if ($this->Owner) {
+            $this->Owner->validate();
+        }
+    }
```

Comment:
Here we validate `Grants` and `Owner` even though they are not required. 

So, if they exist, they have to be valid.

Reply:
good catch